### PR TITLE
[PC-5945] Log ConsultAllOffer

### PR DESCRIPTION
**Pull request by justinemignot**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-5945

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.

## Screenshots
image

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn version:testing` (this will create a commit with a tag)
- then run `git push --follow-tags`

### CODE REVIEW

File: src/features/offer/pages/Offer.test.tsx [hunk 1]
```diff
@@ -68,9 +68,63 @@ describe('<Offer />', () => {
...
+  })
+
+  it('should trigger logEvent "ConsultAllOffer" only once', async () => {
+    const offerPage = await renderOfferPage()
+    const scrollView = offerPage.getByTestId('offer-container')
+    await act(async () => {
+      // 1st scroll to bottom => trigger
+      await scrollView.props.onScroll({ nativeEvent: nativeEventBottom })
+    })
+    expect(logConsultAllOffer).toHaveBeenCalledWith(offerResponseSnap.id)
```
Comment (antoinewg) [inline src/features/offer/pages/Offer.test.tsx:107]:
```suggestion
    expect(logConsultAllOffer).toHaveBeenCalledWith(offerId)
```

après rebase

File: src/libs/analytics.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ enum AnalyticsEvent {
   SEE_MORE_CLICKED = 'SeeMoreClicked',
   BUSINESS_BLOCK_CLICKED = 'BusinessBlockClicked',
   EXCLUSIVITY_BLOCK_CLICKED = 'ExclusivityBlockClicked',
+  CONSULT_ALL_OFFER = 'ConsultAllOffer',
```
[the remark this answers is not in the document]
Comment (justinemignot) [reply src/libs/analytics.ts:17]: done :)

File: src/features/offer/pages/Offer.tsx [hunk 3]
```diff
@@ -42,6 +44,7 @@ const OfferComponent: FunctionComponent = () => {
   } = useRoute<UseRouteType<'Offer'>>()
   const { data: offerResponse } = useOffer({ offerId: dehumanizeId(id) })
   const headerScroll = useRef(new Animated.Value(0)).current
+  const hasSeenAllPage = useRef<boolean>(false)
```
Comment (antoinewg) [inline src/features/offer/pages/Offer.tsx:47]:
```suggestion
  const hasSeenWholePage = useRef<boolean>(false)
```

et pareil pour ConsultWholeOffer, non ?

**Review by antoinewg — COMMENTED** [review]
petite remarque linguistique sinon c'est bon

**Review by antoinewg — APPROVED** [review]

**Pull request merged**